### PR TITLE
*: prune unused dependencies

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Prune unused dependencies and a script to check unused dependencies by [cargo-udeps](https://github.com/est31/cargo-udeps).

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

Close https://github.com/tikv/tikv/issues/5236

### CODE REVIEW

File: scripts/check-udeps.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+#
+# This script makes sure that there is unused dependencies in the tikv
+# dependencies tree. It is suitable to run as part of CI.
+
+cargo install cargo-udeps && cargo udeps
```
Comment (BusyJay) [inline scripts/check-udeps.sh]: How about `which cargo-udeps &>/dev/null || cargo install cargo-udeps && cargo udeps`?

Comment (overvenus) [reply scripts/check-udeps.sh]: cargo does not force install udeps if it is already instead.

Comment (BusyJay) [reply scripts/check-udeps.sh]: It will access network first, which can be slow sometime.

**Comment by overvenus** [comment]
PTAL, thanks!

**Comment by kennytm** [comment]
@overvenus (How) can we cherry-pick this to release-4.0? The unnecessary `criterion-papi` dependency currently blocks efforts to build TiKV on AArch64.

**Comment by BusyJay** [comment]
Master will be merged into release-4.0 at some point. /cc @MyonKeminta

**Comment by kennytm** [comment]
OK thanks.

**Pull request merged**